### DISCUSSION TITLE
add Meter and Battery

### DISCUSSION THREAD

**Discussion by BagheeraHans** [addiejanssen/domoticz-solaredge-modbustcp-plugin#20] [Ideas]
In the solaredge_modbus library the Modbus analysis of meters and storage/battery is available. Is it possible to add these values to the Domoticz plugin?

**Comment by addiejanssen** [comment]
Hi,

When I started developig the plugin, the main purpose was to see what the solar pannels that are attached to the inverter are producing.
Over time, I learned that some of the Solaredge inverters have an ability to attach a power meter and batteries.

In discussion #13 read meter data, Attila Gröpler added a zip file with a modified version of the plugin that enables the usage of a power meter in/attached to the inverter.
I have not included the modifications in the plugin, since I have no way to test it myself.

If that is what you are looking for: give it a spin and let us know if it works.

And if there is somebody out there that is willing to open up their inverter to me (I can share my ip address to restrict access), then I am more than happy to try to integrate this in the plugin.

**Reply by sorinpopa** [reply]
I have a solaredge inverter with meter and I could give you access. It's a single phase SE3000H

**Reply by addiejanssen** [reply]
Hi @sorinpopa,

That would be great!
Let's get in touch to discuss how to do this.
Please contact me via email: me (at) addiejanssen (dot) com

**Comment by pfolcato** [comment]
> In the solaredge_modbus library the Modbus analysis of meters and storage/battery is available. Is it possible to add these values to the Domoticz plugin?
Meter and battery added for a 2 phase inverter
[plugin.zip](https://github.com/addiejanssen/domoticz-solaredge-modbustcp-plugin/files/7807801/plugin.zip)

**Comment by janmeint** [comment]
Hello Addie,
End of August I installed a Solar system with a SolarEdge SE7K-RWS inverter (3 phase) and a LG Chem SERU 6.5 low voltage battery.
I like to thank you for your SolarEdge plugin and this upgrade with power meter and battery readouts.
Best regards,
Jan-Meint

**Comment by PeterVermaas** [comment]
Hello Addie,
I'm trying to get data out my SolarEdge SE3500 with 2 energymeters and 1 Tesla battery into my Domoticz. Now using plugin.2.0.2 . Getting data from Inverter itselves and the 2 meters succeeds, however from Battery not possible. Is there some progress in this item? I can help providing the fieldnames; via example.py it was somehow possible to get the fields and data.
[nieuw1.txt](https://github.com/addiejanssen/domoticz-solaredge-modbustcp-plugin/files/12040448/nieuw1.txt)
